openedx/core/djangoapps/user_api/accounts/helpers.py

```diff
@@ -0,0 +1,56 @@
+"""
+Helper functions for the accounts API.
+"""
+import hashlib
+
+from django.conf import settings
+from django.core.files.storage import FileSystemStorage, get_storage_class
+
+PROFILE_IMAGE_SIZES = {
+    'full': 500,
+    'large': 120,
+    'medium': 50,
+    'small': 30
+}
+PROFILE_IMAGE_FORMAT = 'jpg'
+
+
+def get_profile_image_url_for_user(user, size):
+    """Return the URL to a user's profile image for a given size.
+    Note that based on the value of
+    django.conf.settings.PROFILE_IMAGE_DOMAIN, the URL may be relative,
+    and in that case the caller is responsible for constructing the full
+    URL.
+
+    If the user has not yet uploaded a profile image, return the URL to
+    the default edX user profile image.
+
+    Arguments:
+        user (django.auth.User): The user for whom we're generating a
+        profile image URL.
+
+    Returns:
+        string: The URL for the user's profile image.
+
+    Raises:
+        ValueError: The caller asked for an unsupported image size.
+    """
+    if size not in PROFILE_IMAGE_SIZES.values():
```

Comment:
instead of doing `.values()` on the dict (which creates a new list) each time this is called, instead, create a list from the values as a module variable and check that.

```
$ python -m timeit -s 'v = {"full": 500, "large": 150, "medium": 50, "small": 30}' '1 in v.values()'
1000000 loops, best of 3: 0.256 usec per loop
$ python -m timeit -s 'v = [500, 150, 50, 30]' '1 in v'
10000000 loops, best of 3: 0.115 usec per loop
```